### PR TITLE
Cap update for unconfirmed channels

### LINKED ISSUE #250: Handle updates for unconfirmed channels
Currently a channel is only registered in the PFS once the containing block is confirmed (right now 8 blocks ~ 2mins).

It's likely that during this time a `PFSCapacityUpdate` message is sent. We need to find a way to save this information and apply it once the channel in confirmed.

### PR DESCRIPTION
**Pull request by karlb**
Closes #250.

### CODE REVIEW

File: src/pathfinding_service/service.py [hunk 1]
```diff
@@ -39,6 +40,16 @@
 log = structlog.get_logger(__name__)
 
 
+class DeferMessage(Exception):
+    """Stop processing the message and handle it when the channel has been opened"""
+
+    deferred_message: DeferableMessage
+
+    def __init__(self, deferred_message: DeferableMessage):
```
Comment (palango) [inline src/pathfinding_service/service.py]: This should have a `-> None` annotation, strange that mypy doesn't complain.

File: src/pathfinding_service/service.py [hunk 2]
```diff
@@ -242,7 +253,7 @@ def handle_channel_closed(self, event: ReceiveChannelClosedEvent) -> None:
     def handle_message(self, message: Message) -> None:
         try:
             if isinstance(message, PFSCapacityUpdate):
-                changed_cvs = self.on_pfs_update(message)
+                changed_cvs = self.on_capacity_update(message)
```
Comment (palango) [inline src/pathfinding_service/service.py]: Nice, I forgot to update that one!

File: src/pathfinding_service/service.py [hunk 3]
```diff
@@ -293,13 +305,6 @@ def _validate_pfs_update(self, message: PFSCapacityUpdate) -> TokenNetwork:
         if token_network is None:
```
Comment (palango) [inline src/pathfinding_service/service.py]: I can't comment in the signature, but could you also update the naming there? `_validate_pfs_update` -> `_validate_pfs_capacity_update`?

File: src/pathfinding_service/service.py (hunk 1 above)
Comment (karlb) [reply src/pathfinding_service/service.py]: Mypy does not require a `-> None` annotation for `__init__` unless it has not arguments. In that case an annotation is required to tell mypy whether this method is typed or not. If you like I can add it, but I don't see the benefit, personally.

File: src/pathfinding_service/service.py (hunk 3 above)
Comment (karlb) [reply src/pathfinding_service/service.py]: Good point! Consistent renaming is harder than it sounds.

File: src/pathfinding_service/service.py (hunk 1 above)
Comment (palango) [reply src/pathfinding_service/service.py]: Ah right. Fine for me! 👍

**Pull request merged**